src/test/java/com/alphawallet/attestation/ProofOfKnowledgeTest.java

```diff
@@ -103,4 +103,22 @@ public void TestNegativeEqualityProof() {
     assertFalse(crypto.verifyEqualityProof(com1, com2, newPok));
   }
 
+  @Test
+  public void TestContract() {
+    SmartContract sc = new SmartContract();
+
+    for (int i = 0; i < 30; i++) {
+      ProofOfExponent pok = crypto.computeAttestationProof(BigInteger.TEN);
+      assertTrue(crypto.verifyAttestationRequestProof(pok));
```

Comment:
Furthermore, for the smart contract we should be using the proof of equality, i.e. crypto.computeEqualityProof